.github/workflows/build-python.yml

```diff
@@ -160,28 +160,29 @@ jobs:
           pdm run pip freeze | grep '^pandas'
           pdm run pip freeze | grep -q '^pandas==${{ matrix.pandas_v1 && '1' || '2' }}\.'
 
-      - name: Test code (concurrency)
-        run: pdm test-worker
+      - name: Test code (RAM expensive in tests/models, tests/testing, tests/test_metamorphic_invariance.py)
+        if: ${{ startsWith(matrix.os, 'windows') }}
+        run: pdm test-ram tests/models tests/testing tests/test_metamorphic_invariance.py
 
-      - name: Tests integrations/
-        env:
-          PYTEST_XDIST_AUTO_NUM_WORKERS: ${{ startsWith(matrix.os,'windows-') && 1 || 2 }}
-        run: pdm test-integrations
+      - name: Test code (RAM expensive, /tests/models and /tests/testing)
+        if: ${{ !startsWith(matrix.os, 'windows') }}
+        run: pdm test-ram --memray tests/models tests/testing tests/test_metamorphic_invariance.py
```

Comment:
should we remove the `--memray` by default or you think it’s useful too keep it?

Reply:
In the future: can find some way to mark tests and better distribute them (e.g. avoiding running in parallel certain tests), instead of this manual setting at runtime?

Reply:
I think it can stays. This kind of insights maybe useful
![image](https://github.com/Giskard-AI/giskard/assets/4953724/56969d5c-c988-4433-b9ec-98cd3838d630)

Reply:
For the marking, I did add a pytest mark to not run memory intensive tests in parallel, but still had to split them, else it was too much

maybe we can find a way to run each RAM intensive tests independently with a sub process and kill it everytime ?

Reply:
What is the difference then? The weight of the fixture?

Reply:
No, but when running one test in a process and killing it, all the memory used by the process is going to be freed.
Let's say we use xdist, but recreating a new process every N tests => all the lost memory of our tests will be free as soon as the process is killed.